sonar-php-plugin/src/main/java/org/sonar/plugins/php/PHPRulesDefinition.java

```diff
@@ -45,12 +53,21 @@ public void define(Context context) {
 
     for (NewRule rule : repository.rules()) {
       String metadataKey = rule.key();
-      // Setting internal key is essential for rule templates (see SONAR-6162), and it is not done by AnnotationBasedRulesDefinition from sslr-squid-bridge version 2.5.1:
+      // Setting internal key is essential for rule templates (see SONAR-6162), and it is not done by AnnotationBasedRulesDefinition from
+      // sslr-squid-bridge version 2.5.1:
       rule.setInternalKey(metadataKey);
       rule.setHtmlDescription(readRuleDefinitionResource(metadataKey + ".html"));
       addMetadata(rule, metadataKey);
     }
 
+    boolean shouldSetupSonarLintProfile = sonarRuntimeVersion.isGreaterThanOrEqual(Version.parse("6.0"));
+    if (shouldSetupSonarLintProfile) {
+      Set<String> activatedRuleKeys = PHPProfile.activatedRuleKeys();
+      for (NewRule rule : repository.rules()) {
+        rule.setActivatedByDefault(activatedRuleKeys.contains(rule.key()));
```

Comment:
It would still try to load the rules definition of "Sonar Way". This is a bit hacky and deprecated because we might want to have a different ruleset for SonarLint than "Sonar Way".